src/openqaoa-core/requirements.txt

```diff
@@ -10,4 +10,4 @@ semantic_version>=2.10
 autoray>=0.3.1
 requests
 ipython>=8.2.0
-mitiq
+mitiq[qiskit]
```

Comment:
Is the qiskit extension required? The point of having plugins is that you don't have to install all of the larger packages when you want to use openqaoa. Here we would introduce qiskit in the core package again.

Reply:
TL;DR: `ply` is a runtime requirement of `cirq-core[contrib]`/`mitiq[qiskit]`. We can add`ply` as a direct dependency or use `mitiq[qiskit]`.

---

The need for this originated from a test on notebook `examples/15_Zero_Noise_Extrapolation.ipynb`. This [run](https://github.com/entropicalabs/openqaoa/actions/runs/9411305493/job/25993668893) shows the full stack trace.
The dependency graph essentially is:
`openqaoa-core` -> `mitiq` -> `mitiq_qiskit` -> `cirq-core[contrib]`

I'm open to resolving this any other way you prefer.

Reply:
This is a problem because ideally, we want to have all of our plugins separate, let's say if I want to install `openqaoa-core` on its own, I should be able to use all of its functionalities. 
The issue here is that if you add qiskit dependencies to the core package then it loses its purpose.

Can you try adding `ply` to the requirements locally to see if everything still works? Otherwise I will be in touch with the team that implemented ZNE to see if we can package it better to keep the plugin structure.

Reply:
Well, the issue arose from the example notebook, not any of the package tests.

One more alternative could be to add `mitiq[qiskit]` as a requirement to `openqaoa-qiskit`. This seems logical and would add those required dependencies when using `openqaoa` with `qiskit`.

Let me know if that sounds reasonable, and I will make the change.

Reply:
That sounds reasonable to me let's go with this solution!